src/utils/components/SelectTypeahead/SelectTypeahead.tsx

```diff
@@ -0,0 +1,104 @@
+import React, { FC, ReactNode, Ref, useRef, useState } from 'react';
+
+import {
+  MenuToggleElement,
+  Select,
+  SelectList,
+  SelectOption,
+  SelectOptionProps,
+} from '@patternfly/react-core';
+import { useNetworkingTranslation } from '@utils/hooks/useNetworkingTranslation';
+import { isEmpty } from '@utils/utils';
+
+import Toggle from './Toggle';
+import { filterOptions } from './utils';
+
+type SelectTypeaheadProps = {
+  canCreate?: boolean;
+  id: string;
+  newOptionComponent?: (inputValue: string) => ReactNode;
+  options: SelectOptionProps[];
+  placeholder: string;
+  selected: string;
+  setSelected: (newSelection: null | string) => void;
+};
+
+const SelectTypeahead: FC<SelectTypeaheadProps> = ({
+  canCreate = true,
+  id,
+  newOptionComponent,
+  options,
+  placeholder,
+  selected,
+  setSelected,
+}) => {
+  const { t } = useNetworkingTranslation();
+  const [isOpen, setIsOpen] = useState(false);
+  const [inputValue, setInputValue] = useState<string>(selected || '');
+  const [focusedItemIndex, setFocusedItemIndex] = useState<null | number>(null);
+  const textInputRef = useRef<HTMLInputElement>();
+
+  const selectOptions = inputValue ? filterOptions(options, inputValue) : options;
```

Comment:
how does this work if we already have a selected option, does the filter applied to the original options?

Reply:
yes because the initial `inputValue` value is `selected`.

Reply:
so in case we have a selected value, the user will not see all options in the select, which is not a nice behavior.
PF uses both inputValue and filterValue states, inputValue uses also to the selected value, while filterValue is only meant to show for filtering options. the main difference is that when we initialize the states OR select an option the filterValue is becoming empty or null while we keep the input for inputValue.

Reply:
Oh sorry maybe my explanation was wrong.
If you have something selected and you click on the input, you'll see the other options. 
When you click on the input, the initial text will be the selected project but when you type something different, `inputValue` will be different from `selected` and it will filter based on the input text